examples/HotelAndTravel/AddThingsToDoAd.php

```diff
@@ -0,0 +1,337 @@
+                $options[ArgumentNames::THINGS_TO_DO_CENTER_ACCOUNT_ID]
+                    ?: self::THINGS_TO_DO_CENTER_ACCOUNT_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+            exit(1);
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+            exit(1);
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $thingsToDoCenterAccountId the Things to Do Center account ID
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $thingsToDoCenterAccountId
+    ) {
+        // Creates a budget to be used by the campaign that will be created below.
+        $budgetResourceName = self::addCampaignBudget($googleAdsClient, $customerId);
+        // Creates a Things to do campaign.
+        $campaignResourceName = self::addThingsToDoCampaign(
+            $googleAdsClient,
+            $customerId,
+            $budgetResourceName,
+            $thingsToDoCenterAccountId
+        );
+        // Creates an ad group.
+        $adGroupResourceName =
+            self::addAdGroup($googleAdsClient, $customerId, $campaignResourceName);
+        // Creates an ad group ad.
+        self::addAdGroupAd($googleAdsClient, $customerId, $adGroupResourceName);
+    }
+
+    /**
+     * Creates a new campaign budget in the specified customer account.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @return string the resource name of the newly created budget
+     */
+    private static function addCampaignBudget(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        // Creates a campaign budget.
+        $budget = new CampaignBudget([
+            'name' => 'Interplanetary Cruise Budget #' . Helper::getPrintableDatetime(),
+            'delivery_method' => BudgetDeliveryMethod::STANDARD,
+            // Sets the amount of budget.
+            'amount_micros' => 50000000,
+            // Makes the budget explicitly shared.
```

Comment:
Is this a requirement for Things to do campaigns?

Reply:
Yes. Confirmed.

Reply:
Could you add a comment so readers know `false` is not an option?

Reply:
Done.